src/ControllerWithDataProvider/FrameworkControllerWithDataProvider.php

```diff
@@ -32,6 +32,33 @@ public function getControllerMethod(): string
     public function getArguments(): array
     {
         $frameworksVendorToName = $this->parameterProvider->provideArrayParameter(Option::FRAMEWORKS_VENDOR_TO_NAME);
+        $mergeableVendorList    = $this->parameterProvider->provideArrayParameter(Option::FRAMEWORKS_MERGEABLE_VENDORS);
+        $excludableVendors      = array_reduce(
+            $mergeableVendorList,
+            fn (array $excludableVendors, array $currentList): array => $this->reduceExcludableVendors(
+                $excludableVendors,
+                $currentList
+            ),
+            []
+        );
```

Comment:
I cannot understand what this code does. Could you rephrase is so it's self descriptive with less nesting?

Reply:
Essentially, I'm using the return value of a function call to accomplish a reduction process. The reduction process itself is used to identify vendors that should be excluded from those reported on the home page, due to the fact that their data will be merged under another entry.

I see two ways to clarify what is happening here:

- Inline the function itself (`FrameworkControllerWithDataProvider::reduceExcludableVendors`) here.
- Move the reduction operation into a new function entirely, something along the line of `removeExcludableVendorsFromFrameworkVendorName(array $frameworksVendorToName): array`.

Which would you prefer? Or do you have another suggestion?

Reply:
Thanks. This comment should be embodied in the code itself, so it's not needed to explain it.

> Move the reduction operation into a new function entirely.

This sound good :+1: It can be covered with test as well then.